torch/csrc/byte_order.cpp

```diff
@@ -140,6 +140,17 @@ void THP_decodeHalfBuffer(THHalf* dst, const uint8_t* src, THPByteOrder order, s
   }
 }
 
+void THP_decodeBFloat16Buffer(THBFloat16* dst, const uint8_t* src, THPByteOrder order, size_t len)
+{
+  for (size_t i = 0; i < len; i++) {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+    union { uint16_t x; THBFloat16 f; };
```

Comment:
Technically, type punning via union in C++ is not defined either (once again, use a memcpy)

Reply:
Im not sure if im following. This is identical implementation to Half and they both are using memcpy in decodeUInt16BE() method. what am i missing?

Reply:
See e.g. here https://stackoverflow.com/a/31080901. Apparently Half is also using ub (kinda, it's fine with gcc)